### PR TITLE
Allow unlocking PKCS12 with FORTANIX_PKCS12_PASSPHRASE

### PR DESCRIPTION
**Pull request by zugzwang**

### CODE REVIEW

File: openpgp-dsm/src/lib.rs [hunk 1]
```diff
@@ -1855,8 +1856,17 @@ fn try_unlock_p12(cert_file: String) -> Result<Identity> {
...
-        Ok(id)
+        return Ok(id)
     } else {
+        // Try to unlock with env var passphrase
+        if let Ok(pass) = env::var(ENV_P12_PASS) {
+            if let Ok(id) = Identity::from_pkcs12(&cert, &pass) {
+                info!("unlocked PKCS12 identity with {:?}", ENV_P12_PASS);
+                return Ok(id)
+            } else {
+                info!("could not unlock PKCS12 with {:?}", ENV_P12_PASS);
```
Comment (xinyufort) [inline openpgp-dsm/src/lib.rs:1867]: Nit of nits (again, feel free to ignore): "PKCS12 identity"?

**Review by xinyufort — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -30,7 +30,11 @@ variables need to be set in order to communicate with DSM.
   openssl pkcs12 -export -out identity.pfx -inkey private.key -in public.crt
   ```
   If a password is set for the PKCS12 file, then `sq-dsm` will ask for it on
-  each key usage (which can happen several times on one PGP operation).
+  each key usage (which can happen several times on one PGP operation). See
+  next environment variable.
```
[the remark this answers is not in the document]
Comment (zugzwang) [reply README.md:34]: +1 I was not very inspired

File: README.md [hunk 3]
```diff
@@ -30,7 +30,11 @@ variables need to be set in order to communicate with DSM.
...
+- `FORTANIX_PKCS12_PASSPHRASE`, the passphrase to unlock `identity.pfx` as
+  generated above. If the password is incorrect, `sq-dsm` will ask for it on
```
[the remark this answers is not in the document]
Comment (zugzwang) [reply README.md:36]: Done

File: openpgp-dsm/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (zugzwang) [reply openpgp-dsm/src/lib.rs:1867]: Yes

Comment (zugzwang) [reply openpgp-dsm/src/lib.rs:1867]: Yes

**Pull request merged**